Updates to conform the People page to the styling and layout of Organization and Repository

codycooperross opened the pull request:
## Purpose
<!--- _Describe the problem or feature in addition to a link to the issues._ -->

The Person page styling and layout differs significantly from other pages, like Organizations and Repositories. This PR addresses these styling and layout inconsistencies as well as a few visual bugs. See comparison below:

Old:
<img width="1530" alt="Screen Shot 2025-02-26 at 10 51 01 AM" src="https://github.com/user-attachments/assets/ffd34fd1-4099-4a54-a90d-584db99a6164" />

New:
<img width="1530" alt="Screen Shot 2025-02-26 at 10 51 05 AM" src="https://github.com/user-attachments/assets/15fe3758-0a48-42b5-bf97-4aef7721459d" />

closes: _Add github issue that originated this PR_

## Approach
<!--- _How does this change address the problem?_ -->

#### Open Questions and Pre-Merge TODOs
<!--- - [ ] Use github checklists. When solved, check the box and explain the answer. -->

## Learning
<!--- _Describe the research stage_ -->

<!--- _Links to blog posts, patterns, libraries or addons used to solve this problem_ -->


## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

- [x] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Reviewer, please remember our [guidelines](https://datacite.atlassian.net/wiki/spaces/TEC/pages/1168375809/Pull+Request+Guidelines):

- Be humble in the language and feedback you give, ask don't tell.
- Consider using positive language as opposed to neutral when offering feedback. This is to avoid the negative bias that can occur with neutral language appearing negative.
- Offer suggestions on how to improve code e.g. simplification or expanding clarity.
- Ensure you give reasons for the changes you are proposing.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

- **Refactor**
  - Streamlined the person profile view by removing the display of personal IDs and aggregated metrics.
  - Enhanced metadata presentation with updated reference information for clarity, including a new URL prop.
  - Improved the handling of metrics in the metadata component, allowing for conditional display based on available data.

- **Style**
  - Adjusted layout spacing in related content and secondary profiles for a cleaner, more consistent appearance, including the addition of a new CSS class for subtitles.
  - Removed unnecessary vertical spacing for the `other-profiles` section to refine the layout.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

jrhoads commented:
Overall I like the direction this is going. Thanks for working on this @codycooperross 

- The addition of the Metrics is great.
- Moving the additional metdata to the middle works well.
- Making the achievements responsive is also fantastic.

I think that there are a couple of things to refine.

- There's one part where a URL is passed in to the person metadata that triggers a style change.  This seems like it should be handled another way...
- The Title in the person page links back to the person page.  
  - Other places that have the title, link to an external page (Works -> doi.org..., Organization -> ror.org...,) I think the Person should go to their respective orcid page with similar styling as we see with Works and Orgs.
  - Repository pages should do this as well when available
- Alternative names for a person are listed at the same size as each other.
  - We should probably shrink alternate names like the organization search results do.
  - Organization Detail pages should do this as well
- There's one part where a URL is passed in to the PersonMetadata that triggers a style change.
  - This seems like it should be handled another way but I'm drawing a blank on what to specifically suggest.
  - It may be fine for now but should be revisited

A couple of other things for the future.

- The metrics could be called asynchronously.  Like the Organization Detail pages do.
- We could refactor the accessibility achievements to fetch/render async.  And generalize them for other contexts.

codycooperross commented:
> * The Title in the person page links back to the person page.
>   
>   * Other places that have the title, link to an external page (Works -> doi.org..., Organization -> ror.org...,) I think the Person should go to their respective orcid page with similar styling as we see with Works and Orgs.
>   * Repository pages should do this as well when available

Fixed! 

> * Alternative names for a person are listed at the same size as each other.
>   * We should probably shrink alternate names like the organization search results do.
>   * Organization Detail pages should do this as well

Fixed! 

> * There's one part where a URL is passed in to the PersonMetadata that triggers a style change.
>   
>   * This seems like it should be handled another way but I'm drawing a blank on what to specifically suggest.
>   * It may be fine for now but should be revisited

I agree that this is weird. I was adopting the pattern used by WorkListing: https://github.com/datacite/akita/blob/people-style/src/components/SearchWork/SearchWork.tsx#L52 Not sure if I have any immediate ideas for changing this. 

> A couple of other things for the future.
> 
> * The metrics could be called asynchronously.  Like the Organization Detail pages do.

Good point, and agreed. Same for repositories.

> * We could refactor the accessibility achievements to fetch/render async.  And generalize them for other contexts.

Also agreed.
